pkg/daemons/agent/agent_windows.go

```diff
@@ -2,5 +2,142 @@
 
 package agent
 
-func createRootlessConfig(argsMap map[string]string, hasCfs, hasPIDs bool) {
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/util"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/kubernetes/pkg/kubeapiserver/authorizer/modes"
+)
+
+func checkRuntimeEndpoint(cfg *config.Agent, argsMap map[string]string) {
+	if strings.HasPrefix(argsMap["container-runtime-endpoint"], windowsPrefix) {
+		argsMap["container-runtime-endpoint"] = cfg.RuntimeSocket
+	} else {
+		argsMap["container-runtime-endpoint"] = windowsPrefix + cfg.RuntimeSocket
+	}
+}
+
+func kubeProxyArgs(cfg *config.Agent) map[string]string {
+	argsMap := map[string]string{
+		"proxy-mode":           "kernelspace",
+		"healthz-bind-address": "127.0.0.1",
+		"kubeconfig":           cfg.KubeConfigKubeProxy,
+		"cluster-cidr":         util.JoinIPNets(cfg.ClusterCIDRs),
+	}
+	if cfg.NodeName != "" {
+		argsMap["hostname-override"] = cfg.NodeName
+	}
+
+	argsMap["feature-gates"] = addFeatureGate(argsMap["feature-gates"], "WinOverlay=true")
+
+	if sourceVip := waitForManagementIp("vxlan0"); sourceVip != "" {
+		argsMap["source-vip"] = sourceVip
+	}
+
+	return argsMap
+}
+
+func kubeletArgs(cfg *config.Agent) map[string]string {
+	argsMap := map[string]string{
+		"healthz-bind-address":         "127.0.0.1",
+		"read-only-port":               "0",
+		"cluster-domain":               cfg.ClusterDomain,
+		"kubeconfig":                   cfg.KubeConfigKubelet,
+		"eviction-hard":                "imagefs.available<5%,nodefs.available<5%",
+		"eviction-minimum-reclaim":     "imagefs.available=10%,nodefs.available=10%",
+		"fail-swap-on":                 "false",
+		"authentication-token-webhook": "true",
+		"anonymous-auth":               "false",
+		"authorization-mode":           modes.ModeWebhook,
+	}
+	if cfg.PodManifests != "" && argsMap["pod-manifest-path"] == "" {
+		argsMap["pod-manifest-path"] = cfg.PodManifests
+	}
+	if err := os.MkdirAll(argsMap["pod-manifest-path"], 0755); err != nil {
+		logrus.Errorf("Failed to mkdir %s: %v", argsMap["pod-manifest-path"], err)
+	}
+	if cfg.RootDir != "" {
+		argsMap["root-dir"] = cfg.RootDir
+		argsMap["cert-dir"] = filepath.Join(cfg.RootDir, "pki")
+		argsMap["seccomp-profile-root"] = filepath.Join(cfg.RootDir, "seccomp")
```

Comment:
is this valid?

Reply:
What we currently have works and I don't stop these from getting applied.

Reply:
@brandond happy to remove if you want, otherwise, I think this may be good.